crates/core/tedge/src/cli/connect/c8y_direct_connection.rs

```diff
@@ -121,11 +122,42 @@ fn read_pvt_key(
 ) -> Result<rustls_0_19::PrivateKey, ConnectError> {
     // Become BROKER_USER to read the private key
     let _user_guard = user_manager.become_user(tedge_users::BROKER_USER)?;
-    let f = File::open(key_file)?;
+    match parse_pvt_key(key_file.clone()) {
+        Ok(key) => Ok(key),
+        Err(ConnectError::UnsupportedPvtKeyFormat) => match parse_pvt_rsa_key(key_file) {
+            Ok(key) => Ok(key),
+            Err(e) => Err(e),
+        },
+        Err(e) => Err(e),
+    }
+}
+
+fn parse_pvt_key(
```

Comment:
Since this function is looking for a very specific file format, I would make this explicit in the name.

Why not `parse_pkcs8_key` ?

Reply:
fixed